File: CrackingTheCodingInterview/TechnicalQuestions/LongestSubstring.java
public int lengthOfLongestSubstring(String s) {
    int longest = 0;
    int i = 0, j = 0;
    Set<Character> set = new HashSet<>();
    while (i < s.length()) {
        if (set.contains(s.charAt(i))) {
            set.remove(s.charAt(j));
            j++;
        } else {
            set.add(s.charAt(i));
            i++;
        }
        longest = Math.max(set.size(), longest);
    }
    return longest;
}

File: CrackingTheCodingInterview/TechnicalQuestions/LongestSubstring.js
var lengthOfLongestSubstring = function(s) {
    let i = 0, j = 0;
    let longest = 0;
    let set = new Set();
    while (i < s.length && j < s.length) {
        if (set.has(s[i])) {
            set.delete(s[j]);
            j++;
        } else {
            set.add(s[i]);
            i++;
            longest = Math.max(set.size, longest);
        }
    }
    return longest;
};